### PR TITLE
Support Just task runner

### PR DESCRIPTION
**Pull request by hyperupcall**
This adds support for [Just](https://github.com/casey/just), a command/task runner. The syntax is very similar to Make.

I'm not sure if I should extract the "run Tox integration" tests out to a different PR - I had to do it in #171 as well. Feel free to let me know if you think I should do so :)

### CODE REVIEW

File: src/mk/tools/just.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    def run(self, action: Action | None = None) -> None:
+        cmd = ["just"]
+        if action:
+            cmd.append(action.name)
+        run_or_fail(cmd, tee=True)
+
+    def is_present(self, path: Path) -> bool:
+        for name in ["Justfile", "justfile", "JUSTFILE", ".justfile"]:
+            justfile = os.path.join(path, name)
+            if os.path.isfile(justfile):
```
Comment (gotmax23) [inline src/mk/tools/just.py:27]:
```suggestion
            justfile = path / name
            if justfile.is_file():
```

Comment (hyperupcall) [reply src/mk/tools/just.py:27]: This file was written as similarly to `make.py` as possible. If this is needed, I think another PR could convert more of the old-style path things at once?

**Review by ssbarnea — APPROVED** [review]

**Comment by ssbarnea** [comment]
@hyperupcall Can you please check the checkmark about allowing pushes from core, I wanted to make some fixes to the branch.

**Comment by ssbarnea** [comment]
@hyperupcall @gotmax23 I sent invites to both of you on https://github.com/pycontribs/mk/invitations

**Comment by ssbarnea** [comment]
@hyperupcall I created a branch for this, feel free to push to it so we can bring it to ready. We need test coverage as current implementation does not seem to be working fine. I tried it myself with `just` itself and it failed to list available targets.

**Pull request closed without merging**